src/Model.jl

```diff
@@ -94,28 +96,43 @@ function Model(lattice::AbstractMatrix{T},
                atoms::Vector{<:Element}=Element[],
                positions::Vector{<:AbstractVector}=Vec3{T}[];
                model_name="custom",
-               n_electrons::Int=sum(n_elec_valence, atoms; init=0),
+               εF=nothing,
+               n_electrons=isnothing(εF) ? n_electrons_from_atoms(atoms) : nothing,
```

Comment:
I really would like the type annotation here to keep people from using floating-point electrons.